File: chinko_gen_py/chinko_gen_py/chinko_generator.py
import random

class ChinkoGenerator:
    _is_chinko = [0, 1, 2]
    _chinko_dict = ["ち", "ん", "こ"]

    def __init__(self) -> None:
        self._chinko_list = []

    def __str__(self) -> str:
        word = [self._chinko_dict[int(elm)] for elm in self._chinko_list]
        return "".join(word)
    
    def gen_triplet(self):
        triplet = [random.randint(0, 2) for _ in range(3)]
        self._chinko_list.extend(triplet)
        return triplet == self._is_chinko
    

            

File: chinko_gen_rs/src/chinko_generator.rs
use rand::Rng;

pub struct ChinkoGenerator {
    is_chinko: [u8; 3],
    chinko_dict: [&'static str; 3],
    chinko_list: Vec<u8>,
}

impl ChinkoGenerator {
    pub fn new() -> ChinkoGenerator {
        ChinkoGenerator {
            is_chinko: [0, 1, 2],
            chinko_dict: ["ち", "ん", "こ"],
            chinko_list: Vec::new(),
        }
    }

    pub fn gen_triplet(&mut self) -> bool {
        let mut rng = rand::thread_rng();
        let triplet: Vec<u8> = (0..3).map(|_| rng.gen_range(0..=2)).collect();
        self.chinko_list.extend_from_slice(&triplet);
        triplet == self.is_chinko
    }
}

impl std::fmt::Display for ChinkoGenerator {
    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
        let word: String = self
            .chinko_list
            .iter()
            .map(|elm| self.chinko_dict[*elm as usize].to_string())
            .collect();
        write!(f, "{}", word)
    }
}

// fn main() {
//     let mut generator = ChinkoGenerator::new();
//     generator.gen_triplet();
//     println!("{}", generator);
// }
